src/graph/entries.mo

```diff
@@ -0,0 +1,37 @@
+import Array "mo:stdlib/array.mo";
+import Blob "mo:stdlib/blob.mo";
+import HashMap "mo:stdlib/hashMap.mo";
+import Hash "mo:stdlib/hash.mo";
+import Iter "mo:stdlib/iter.mo";
+import List "mo:stdlib/list.mo";
+import Nat "mo:stdlib/nat.mo";
+import Option "mo:stdlib/option.mo";
+
+import Types "./types.mo";
+
+module {
+  type Entry = Types.Entry;
+  type EntryId = Types.EntryId;
+
+  public class Entries () {
+
+    func natEq (x : Nat32, y : Nat32) : Bool { x == y };
+    func hashEntryId (x : EntryId) : Hash.Hash { Nat.toWord32(Nat.fromNat32(x)) };
+    let hashMap = HashMap.HashMap<EntryId, Entry>(1, natEq, hashEntryId);
+
+    public func addConnection (fromUser : EntryId, toUser : EntryId) : () {
+      var existing = Option.unwrap<Entry>(hashMap.get(fromUser));
```

Comment:
Can you write this with an explicit `switch`, where you handle the `null` case with some behavior, or the indication of an error?

FWIW, @rossberg wants to remove Option.unwrap from the standard library to make this idiom (no explicit failure code) even harder to write.

Reply:
> Can you write this with an explicit switch, where you handle the null case with some behavior, or the indication of an error?

Yes, I actually tried that first and couldn't get it working. I'll post what I tried and the error I received and maybe you can suggest changes.

Reply:
This works, but still uses unwrap...
```
    public func addConnection (fromUser : EntryId, toUser : EntryId) : () {
      var maybeEntry = hashMap.get(fromUser);
      switch (maybeEntry) {
        case (null) {
          let entry : Entry = {
            id = fromUser;
            connections = [toUser];
            invitations = [];
          };
          ignore hashMap.set(fromUser, entry);
        };
        case (_) {
          let existing = Option.unwrap<Entry>(maybeEntry);
          let entry : Entry = {
            id = existing.id;
            connections = Array.append<EntryId>(existing.connections, [toUser]);
            invitations = existing.invitations;
          };
          ignore hashMap.set(fromUser, entry);
        };
      };
    };
```
If I don't unwrap I get that an object is expected instead of type `?Entry/1`.

Reply:
```
case (?existing) {
  let entry : …
```
should do what you want to do.

Reply:
That works, great. Change coming.